File: Stack/기능개발.js
// 기능 개발
// https://programmers.co.kr/learn/courses/30/lessons/42586?language=javascript

function solution(progresses, speeds) {
    let max = 0;
    let stk = [];
    const result = [];
    for(let i = 0; i < progresses.length; i++ ){
        const day = Math.ceil((100 - progresses[i]) / speeds[i]);
		if (!stk.length) max = day;      
		if (max < day) {
			result.push(stk.length);
			stk = []
			max = day;
		}
		stk.push(day);
		if(i === progresses.length - 1)
			result.push(stk.length);
    }
    return result;
}

File: Stack/기능개발.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int jobs = 1;
        double max = Math.ceil((100 - progresses[0]) / (double) speeds[0]);
        List<Integer> answer = new ArrayList<Integer>();
        
        for(int i = 1; i < progresses.length; i++){
            double day = Math.ceil((100 - progresses[i]) / (double) speeds[i]);
            if(max < day){
                max = day;
                answer.add(jobs);
                jobs = 1;
            }else{
                jobs++;
            }
        }
        answer.add(jobs);
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}
